include/EquationSystem.h

```diff
@@ -224,12 +224,15 @@ class EquationSystem
   void report_built_supp_alg_names();
   bool supp_alg_is_requested(std::string name);
   bool supp_alg_is_requested(std::vector<std::string>);
-
+  
+  void reset_nonlinear_iteration_count();
   bool nodal_src_is_requested();
 
   EquationSystems &equationSystems_;
   Realm &realm_;
   std::string name_;
+  std::string sysName_;
+  bool reportMyResiduals_;
```

Comment:
Can you explain why using `linsys_->provideOutput_` within EquationSystem methods was not possible? Should this variable be removed from the code now that we have `reportMyResiduals_`?